docs/concepts/resolution.md

```diff
@@ -235,18 +235,20 @@ available versions for a transitive dependency without adding a direct requireme
 
 ## Dependency overrides
 
-Sometimes, the requirements defined by a dependency are too strict, and a working version of a
-package is not allowed (often, causing resolution to fail). uv allows overriding requirements
-defined by dependencies to unblock resolution.
+Overrides allow bypassing failing or undesirable resolutions by overriding the declared dependencies
+of a package. Overrides are a useful last resort for cases in which the you know that a dependency
+is compatible with a newer version of a package than the package declares, but the package has not
```

Comment:
"compatible with a newer version of a package than the package declares"

This is confusing because "newer version of a package" is referring to the thing it depends on, but "than the package declares" is referring to the thing with the dependency

Reply:
```suggestion
is compatible with a newer version of a package than the dependency declares, but the dependency has not
```

Reply:
Maybe?